compiler/rustc_codegen_ssa/src/lib.rs

```diff
@@ -190,3 +191,33 @@ pub fn looks_like_rust_object_file(filename: &str) -> bool {
     // Check if the "inner" extension
     ext2 == Some(RUST_CGU_EXT)
 }
+
+const RLINK_MAGIC: &'static [u8; 5] = b"rlink";
```

Comment:
Maybe use `rustlink`? Also I think it should contain a version number as we have changed the encoded format of strings before.